### DISCUSSION TITLE
Is it possible to have 2 different installations of the same formula?

### DISCUSSION THREAD

**Discussion by maichai** [Homebrew/discussions#1725] [Everyday usage]
I would like to have 2 installations of the proxmark3 formula, each for a different model of the Proxmark3 device:

1. brew install --HEAD --with-blueshark proxmark3
2. brew install --HEAD --with-generic proxmark3

It would be great to be able to use the right pm3 client for each device without needing to uninstall the other client version and re-install/compile the right version.

**Comment by carlocab** [comment]
Unfortunately not. What you can do is extract one of these into a separate formula. To do this, you'll need to create a tap (see `brew help tap-new`). When you have your tap, do `brew extract proxmark3 maichai/tap-name`. This will create a versioned copy of `proxmark3` in your own tap, which may wish to call `proxmark3-blueshark` or `proxmark3-generic`.

Here are some docs to help you get started on this:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/
